presto-main/src/main/java/com/facebook/presto/connector/system/jdbc/TypesJdbcTable.java

```diff
@@ -0,0 +1,115 @@
+import com.facebook.presto.spi.RecordCursor;
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.spi.connector.ConnectorTransactionHandle;
+import com.facebook.presto.spi.predicate.TupleDomain;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+
+import static com.facebook.presto.connector.system.jdbc.ColumnJdbcTable.columnSize;
+import static com.facebook.presto.connector.system.jdbc.ColumnJdbcTable.jdbcDataType;
+import static com.facebook.presto.connector.system.jdbc.ColumnJdbcTable.numPrecRadix;
+import static com.facebook.presto.metadata.MetadataUtil.TableMetadataBuilder.tableMetadataBuilder;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DateType.DATE;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.RealType.REAL;
+import static com.facebook.presto.spi.type.SmallintType.SMALLINT;
+import static com.facebook.presto.spi.type.TimeType.TIME;
+import static com.facebook.presto.spi.type.TimeWithTimeZoneType.TIME_WITH_TIME_ZONE;
+import static com.facebook.presto.spi.type.TimestampType.TIMESTAMP;
+import static com.facebook.presto.spi.type.TimestampWithTimeZoneType.TIMESTAMP_WITH_TIME_ZONE;
+import static com.facebook.presto.spi.type.TinyintType.TINYINT;
+import static com.facebook.presto.spi.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.spi.type.VarcharType.createUnboundedVarcharType;
+
+public class TypesJdbcTable
+        extends JdbcTable
+{
+    public static final SchemaTableName NAME = new SchemaTableName("jdbc", "types");
+
+    public static final ConnectorTableMetadata METADATA = tableMetadataBuilder(NAME)
+            .column("type_name", createUnboundedVarcharType())
+            .column("data_type", BIGINT)
+            .column("precision", BIGINT)
+            .column("literal_prefix", createUnboundedVarcharType())
+            .column("literal_suffix", createUnboundedVarcharType())
+            .column("create_params", createUnboundedVarcharType())
+            .column("nullable", BIGINT)
+            .column("case_sensitive", BOOLEAN)
+            .column("searchable", BIGINT)
+            .column("unsigned_attribute", BOOLEAN)
+            .column("fixed_prec_scale", BOOLEAN)
+            .column("auto_increment", BOOLEAN)
+            .column("local_type_name", createUnboundedVarcharType())
+            .column("minimum_scale", BIGINT)
+            .column("maximum_scale", BIGINT)
+            .column("sql_data_type", BIGINT)
+            .column("sql_datetime_sub", BIGINT)
+            .column("num_prec_radix", BIGINT)
+            .build();
+
+    private static final Map<Type, String> SQL_TYPES = ImmutableMap.<Type, String>builder()
```

Comment:
This seems to be missing some types: varchar, array, map, row, etc. How about injecting `TypeManager` and getting all types? I think the JDBC driver should support fetching any type (though the user of the driver has to know what to do with it).